src/stories/Inputs/Upload/Upload.stories.tsx

```diff
@@ -27,4 +27,37 @@ export default {
 const Template: ComponentStory<typeof Upload> = (args) => <Upload {...args} />;
 
 export const Default = Template.bind({});
-Default.args = {};
+Default.args = {
+  isUploading: false,
+  onChange: () => {},
+};
+
+export const Uploading = Template.bind({});
+Uploading.args = {
+  ...Default.args,
+  isUploading: true,
```

Comment:
These various states that deal with the `HTTP` lifecycle will be controlled by the calling parent component. Many times we will consider handling these concerns as side effects in the component itself, often times reaching for the `useEffect` hook. I prefer to handle that in a parent component and make the props that I'm passing to render components very explicit. This seems to play well with how Storybook works too.

Reply:
So if we were using the `Upload` component as part of the `ProfilePage` it might look something like this:

```
  const handleUpload = () => {
    // some function that handles our HTTP request to upload the profile picture 
   // returns our `isUpload` and `isUploadSuccessful` | `hasFailed` state 
  }
<ProfilePage>
  {isEdit && (
      <Upload 
          isUploading={isUploading}
          isUploadSuccessful={isUploadSuccessful}
          hasFailed={hasFailed}
     />
  )}
</ProfilePage>
```

The alternative would be to perform our HTTP request as part of the `Upload` component itself and not have any props and just set state internally. While this is perfectly valid I think that it does a bad job of separating concerns. It also makes testing harder and would be really difficult to implement the different versions of `Upload` in storybook because it doesn't expose any props.